### PR TITLE
Fix pruning group conv2d

### PR DESCRIPTION
**Pull request by wanghaoshuang**
PaddleClas测试结果：
https://github.com/PaddlePaddle/PaddleSlim/issues/721

### CODE REVIEW

**Comment by wanghaoshuang** [comment]
@minghaoBD

File: paddleslim/dygraph/prune/filter_pruner.py [hunk 1]
```diff
@@ -291,73 +289,58 @@ def restore(self):
...
         Returns:
             plan: An instance of PruningPlan that can be applied on model by calling 'plan.apply(model)'.
 
         """
+        pruned_axis = pruned_axis[0] if isinstance(pruned_axis,
+                                                   list) else pruned_axis
+        assert (isinstance(pruned_axis, int))
         if var_name in self.skip_vars:
             _logger.warn(
                 f"{var_name} is skiped beacause it is not support for pruning derectly."
```
Comment (minghaoBD) [inline paddleslim/dygraph/prune/filter_pruner.py]: support -> supported
derectly -> directly

File: paddleslim/dygraph/prune/filter_pruner.py [hunk 2]
```diff
@@ -291,73 +289,58 @@ def restore(self):
...
-                        'value': np.array([]),
-                        'pruned_dims': [1],
-                      }
-        }
-        """
+    def cal_mask(self, pruned_ratio, collection):
         raise NotImplemented("cal_mask is not implemented")
 
-    def prune_var(self, var_name, pruned_dims, pruned_ratio, apply="impretive"):
+    def prune_var(self, var_name, pruned_axis, pruned_ratio, apply="impretive"):
```
Comment (minghaoBD) [inline paddleslim/dygraph/prune/filter_pruner.py:295]: apply在参数描述中没有体现。

File: paddleslim/prune/collections.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+class PruningCollection(object):
+    """
+    A group of pruning operations.
+    
+      conv1-->conv2-->batch_norm
+
+    For the network defined above, if weight of conv1 is pruned on 0-axis,
+    weight of'conv2' should be pruned on 1-axis. The pruning operations on 0-axis of
+    'conv1' and that on 1-aixs of 'conv2' is a collection. And the {'name': conv1.weight_name, 'axis': 0}
```
Comment (minghaoBD) [inline paddleslim/prune/collections.py]: that on 1-aixs of 'conv2' -> those on ...

File: paddleslim/prune/collections.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    def __init__(self):
+        self._collections = None
+
+    def __iter__(self):
+        return iter(self._collections)
+
+    def create_pruning_collections(self,
+                                   params,
+                                   graph,
+                                   visited=None,
```
Comment (minghaoBD) [inline paddleslim/prune/collections.py]: visited在参数描述中没有体现。

File: paddleslim/prune/collections.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        if visited is None:
+            visited = {}
+        collections = []
+        unsupported_warnings = set()
+        for _param in params:
+            pruned_params = []
+            param = graph.var(_param)
+            if param is None:
+                _logger.warning(
+                    f"Cann't found relative variables of {_param} because {_param} is not in target program or model. Please make sure {_param} is in your program if you are using static API of PaddlePaddle. And make sure your model in correctly mode and contains {_param} if you are using dynamic API of PaddlePaddle."
```
Comment (minghaoBD) [inline paddleslim/prune/collections.py]: Cann't found -> Couldn't find
correctly -> correct

File: paddleslim/prune/idx_selector.py [hunk 6]
```diff
@@ -26,75 +26,80 @@
...
 @IDX_SELECTOR.register
-def default_idx_selector(group, ratio):
-    """Get the pruned indexes by given ratio.
+def default_idx_selector(group, scores, ratios):
+    """Get the pruned indexes by scores of master tensor.
 
     This function return a list of parameters' pruned indexes on given axis.
-    Each element of list is a tuple with format (name, axis, indexes) in which 'name' is parameter's name
-    and 'axis' is the axis pruning on and `indexes` is indexes to be pruned.
+    Each element of list is a tuple with format (name, axis, indexes)
```
Comment (minghaoBD) [inline paddleslim/prune/idx_selector.py]: indexes -> indices
以及其他地方的indexes

File: paddleslim/prune/idx_selector.py [hunk 7]
```diff
@@ -26,75 +26,80 @@
...
-                    .. code-block: text
-
-                       [("conv1_weights", 0, [0.7, 0.5, 0.6]), ("conv1_bn.scale", 0, [0.1, 0.2, 0.4])]
-
-                    The shape of "conv1_weights" is `[out_channel, in_channel, filter_size, filter_size]`, so
-                    `[0.7, 0.5, 0.6]` are the importance sores of each output channel in "conv1_weights"
-                    while axis is 0.
+       group(Group): A group of pruning operations.
+       scores(dict): The key is name of tensor, the value is a dict with axis as key and scores as value.
+       ratios(doct): The pruned ratio of each tensor. The key is name of tensor and the value is the pruned ratio.
```
Comment (minghaoBD) [inline paddleslim/prune/idx_selector.py]: doct -> dict
以及其他地方的doct

File: paddleslim/dygraph/prune/filter_pruner.py (hunk 1 above)
Comment (wanghaoshuang) [reply paddleslim/dygraph/prune/filter_pruner.py]: Fixed.

File: paddleslim/dygraph/prune/filter_pruner.py (hunk 2 above)
Comment (wanghaoshuang) [reply paddleslim/dygraph/prune/filter_pruner.py:295]: Fixed.

File: paddleslim/prune/collections.py (hunk 3 above)
Comment (wanghaoshuang) [reply paddleslim/prune/collections.py]: Fixed.

File: paddleslim/prune/collections.py (hunk 4 above)
Comment (wanghaoshuang) [reply paddleslim/prune/collections.py]: Fixed.

File: paddleslim/prune/collections.py (hunk 5 above)
Comment (wanghaoshuang) [reply paddleslim/prune/collections.py]: Fixed.

File: paddleslim/prune/idx_selector.py (hunk 6 above)
Comment (wanghaoshuang) [reply paddleslim/prune/idx_selector.py]: Fixed.

File: paddleslim/prune/idx_selector.py (hunk 7 above)
Comment (wanghaoshuang) [reply paddleslim/prune/idx_selector.py]: Fixed.

**Comment by minghaoBD** [comment]
LGTM

**Review by minghaoBD — APPROVED** [review]

**Pull request merged**